### ISSUE TITLE
BUG: After Map Change, Server Gracefully Exits due to Unable to Insert Player into GameMode

### ISSUE THREAD

**Issue by proof88** [proof88/PRooFPS-dd#267]
I see the following server-side logs:

```
PRooFPSddPGE::handleUserConnected(): new remote user Player (connHandleServerSide: 19961977) connected (from 192.168.1.46:35905) and I'm server
  PRooFPSddPGE::handleUserSetupFromServer(): new user Player (connHandleServerSide: 2605500427; IP: 192.168.1.23:56822) connected
  PRooFPSddPGE::handleUserSetupFromServer(): new user Player (connHandleServerSide: 511855507; IP: 192.168.1.135:64257) connected
  PRooFPSddPGE::handleUserSetupFromServer(): failed to insert player Player into GameMode!
```

Somehow all users try to connect with the same name even though everybody changed their name in config file (default.cfg), and anyway server supposed to generate a unique name in such case, somehow player insertion fails.

**Comment by proof88** [comment]
RCA: this issue happens because the unique name is generated in handleUserConnected(), but insertion into m_mapPlayers happens in handleUserSetupFromServer(). Unique name gepends on a properly filled m_mapPlayers. Thus either we should insert in handleUserConnected(), or delay the unique name generation to handleUserSetupFromServer(), or generate unique name at both places.

**Comment by proof88** [comment]
Fixed in v0.1.6.1.

RCA: unique player name for clients was generated in handleUserConnected() which made sure the name was unique considering players already present in m_mapPlayers. However, adding the new player to m_mapPlayers was being done in handleUserSetupFromServer().
In a sunny weather scenario any user connecting to the server generated a handleUserConnected() and handleUserSetupFromServer() sequence in this order.
However, with multiple players reconnecting to the server in burst fashion, 2 MsgUserConnected messages might had been injected after each other into server's msg queue, that could lead to thinking both names are unique even though they were the same.

FIX: fix was to move the code responsible for unique names into handleUserNameChange() which definitely happens for any user after the MsgUserConnected and MsgUserSetupFromServer messages have been already processed for that user.
This also requires allowing a player to be inserted into m_mapPlayers with empty name.
From now on we insert a player into gameMode in MsgUserNameChange because that is the exact point when names become unique.
